docs/source/llms/llm-evaluate/index.rst

```diff
@@ -0,0 +1,538 @@
+    )
+
+Now let's define the ``professionalism`` metric, you will see how each field is set up.
+
+.. code-block:: python
+
+    professionalism = mlflow.metrics.make_genai_metric(
+        name="professionalism",
+        definition=(
+            "Professionalism refers to the use of a formal, respectful, and appropriate style of communication that is "
+            "tailored to the context and audience. It often involves avoiding overly casual language, slang, or "
+            "colloquialisms, and instead using clear, concise, and respectful language."
+        ),
+        grading_prompt=(
+            "Professionalism: If the answer is written using a professional tone, below are the details for different scores: "
+            "- Score 0: Language is extremely casual, informal, and may include slang or colloquialisms. Not suitable for "
+            "professional contexts."
+            "- Score 1: Language is casual but generally respectful and avoids strong informality or slang. Acceptable in "
+            "some informal professional settings."
+            "- Score 2: Language is overall formal but still have casual words/phrases. Borderline for professional contexts."
+            "- Score 3: Language is balanced and avoids extreme informality or formality. Suitable for most professional contexts. "
+            "- Score 4: Language is noticeably formal, respectful, and avoids casual elements. Appropriate for formal "
+            "business or academic settings. "
+        ),
+        examples=[professionalism_example_score_2, professionalism_example_score_4],
+        model="openai:/gpt-3.5-turbo-16k",
+        parameters={"temperature": 0.0},
+        aggregations=["mean", "variance"],
+        greater_is_better=True,
+    )
+
+..
+    TODO(prithvi): add best practice for creating GenAI metrics.
+
+
+Create Per-row LLM Evluation Metrics (Category 2)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is very similar to creating a custom traditional metrics, with the exception of returning a `EvaluationResult` instance.
+Basically you need to:
+
+1. Implement a ``eval_fn`` to define your scoring logic, it must take in 3 args ``predictions``, ``targets`` and ``metrics``.
+   ``eval_fn`` must return a :py:func:`mlflow.metrics.MetricValue` instance.
+2. Pass ``eval_fn`` and other arguments to ``mlflow.metricsmake_metric`` API to create the metric. 
+
+The following code creates a dummy per-row metric called ``"over_10_chars"``: if the model output is greater than 10, 
+the score is 1 otherwise 0.
+
+.. code-block:: python
+
+    def eval_fn(predictions, targets, metrics):
+        scores = []
+        for i in range(len(predictions)):
+            if len(predictions[i]) > 10:
+                scores.append(1)
+            else:
+                scores.append(0)
+        return MetricValue(
+            scores=scores,
+            aggregate_results=standard_aggregations(scores),
+        )
```

Comment:
```suggestion
        scores = []
        for pred in predictions:
            scores.append(len(pred) > 10)
        return MetricValue(
            scores=scores,
            aggregate_results=standard_aggregations(scores),
        )
```

Reply:
hmm I think that gives True/False, although it is compatible with int, I don't want to leave users an impression that they can use boolean as the return value.